app/models/asset.rb

```diff
@@ -52,6 +52,14 @@ class Asset
     event :scanned_infected do
       transition any => :infected
     end
+
+    event :upload_success do
+      transition %i(clean uploaded not_uploaded) => :uploaded
```

Comment:
Do we definitely need to be able to transition from `uploaded` to `uploaded`? Presumably if we were to upload a new file for the `Asset` then we'd also reset the state back to `unscanned` which would mean it transitions through `clean` to `uploaded`.

Reply:
In general, I was leaning towards not being too restrictive on the state transitions, because I didn't want an unforeseen transition to cause the app to start failing. Although I wanted to be more restrictive than the completely unconstrained transitions into `clean` and `infected` states.

In this case, I was thinking that in the past we've run Rake tasks to re-upload the same files, i.e. re-do the upload with exactly the same file as before.

What are the benefits of preventing the transition from `uploaded` to `uploaded`?

Reply:
> In this case, I was thinking that in the past we've run Rake tasks to re-upload the same files, i.e. re-do the upload with exactly the same file as before.

Good point. Although this isn't going to be possible as soon as we remove the files from NFS.

> What are the benefits of preventing the transition from `uploaded` to `uploaded`?

I guess I think it's a bit easier to understand if we can only transition from `clean` to `uploaded`.

Reply:
OK. I'll go with that approach. Thanks.